### ISSUE TITLE
Error when sanitize strategy escaped and value is undefined. 

### ISSUE THREAD

**Issue by imamovic** [angular-translate/angular-translate#1044]
Error in angular-translate.js on line 284
Error: TypeError: angular.element(...).text(...).html is not a function

```javascript
var htmlEscapeValue = function (value) { 
        return angular.element('<div></div>').text(value).html();
  };
```

Suggestion to check type of value before call: return angular.element('<div></div>').text(value).html();

**Comment by knalli** [comment]
Thanks for reporting. I'm still wonder why this can be null actually. `text` is a defined?!

**Comment by imamovic** [comment]
Problem is when model is not populated with data and property of model is undefined.

**Comment by knalli** [comment]
Which model? This particular piece of code does not depend on any model.

* `angular.element('<div></div>').text(undefined).html() == ''`
* `angular.element('<div></div>').text(null).html() == 'null'`
* `angular.element('<div></div>').text('').html() == ''`

Still not getting it.

**Comment by imamovic** [comment]
Here is full error trace: 

```error

$translateSanitizationProvider/htmlEscapeValue()angular...late.js (line 283)
$translateSanitizationProvider/mapInterpolationParameters()angular...late.js (line 307)
$translateSanitizationProvider/mapInterpolationParameters/<()angular...late.js (line 300)
forEach()angular.js (line 331)
$translateSanitizationProvider/mapInterpolationParameters()angular...late.js (line 298)
$translateSanitizationProvider/strategies.escapeParameters()angular...late.js (line 137)
$translateSanitizationProvider/this.$get</applyStrategies/<()angular...late.js (line 213)
forEach()angular.js (line 323)
$translateSanitizationProvider/this.$get</applyStrategies()angular...late.js (line 209)
$translateSanitizationProvider/this.$get</<.sanitize()angular...late.js (line 277)
$translateDefaultInterpolation/$translateInterpolator.interpolate()angular...late.js (line 2365)
$translate/this.$get</determineTranslation()angular...late.js (line 1721)
$translate/this.$get</$translate()angular...late.js (line 1331)
linkFn/updateTranslation()angular...late.js (line 2633)
linkFn/updateTranslations()angular...late.js (line 2625)
$RootScopeProvider/this.$get</Scope.prototype.$digest()angular.js (line 14308)
$RootScopeProvider/this.$get</Scope.prototype.$apply()angular.js (line 14571)
done()angular.js (line 9698)
completeRequest()angular.js (line 9888)
requestLoaded()angular.js (line 9829)
	

return angular.element('<div></div>').text(value).html();

angular.js (line 11655)

```

Problem is when in function  "mapInterpolationParameters" is send object
```javascript
var object = {
   field: "name"
   message: "{javax.validation.constraints.NotNull.message}"
   type: "Operator"
   valid: false
   validator:"required"
   value: undefined
}
```
It go through every parameters and call function htmlEscapeValue(value). When get on property "value" error is happened because value= undefined. 

Value is undefined because form user valdr (plugin for validation) and object used on form is Angular resource object. When new resource object is created value are undefined.

**Comment by knalli** [comment]
>  When get on property "value" error is happened because value= undefined.

But that conflicts with my "test" mentioned in the [last post](https://github.com/angular-translate/angular-translate/issues/1044#issuecomment-105445360): `angular.element('<div></div>').text(undefined).html() == ''`

**Comment by knalli** [comment]
Please demonstrate you case, as minimal as possible. You can use [this](http://plnkr.co/edit/cn1cGtsHqJTGB25OFV06) here as base..

**Comment by imamovic** [comment]
Test this "angular.element('<div></div>').text(undefined).html()" without included jquery. 

If you do not have included jquery, this: angular.element('<div></div>').text(undefined).html() throw error. 

If you have included jquery, this : angular.element('<div></div>').text(undefined).html() return "". 

Now this is not related to this plugin (but it will be good if you fix this or write jquery as dependency), it is problem in angular jqLite. (behavior in jquery and jqLite is not same). 

Please write can we expect your solution without jquery dependency.

**Comment by knalli** [comment]
Ah, jqLIte. Right, okay, thanks for the hint. The next release will contain a patch.

**Comment by mathroule** [comment]
Same issue without jQuery included.

**Comment by knalli** [comment]
landed as ef1b10a

**Comment by knalli** [comment]
Fixed in 2.7.2 (released two days ago).
